src/bridges/notional/NotionalBridge.sol

```diff
@@ -0,0 +1,207 @@
+            IRollupProcessor(ROLLUP_PROCESSOR).receiveEthFromBridge{value: outputValueA}(_interactionNonce);
+        }
+    }
+
+    /**
+     * @notice Helper functon to compute the fcash amount
+     * @dev If the token is a ctoken, then we need to convert the amount to its underlying in 8 decimals
+     * @dev If the token is an underlying token, then we simply scale it to 8 decimals
+     * @param _token is the address of input asset token
+     * @param _amount is the amount of input asset token
+     * @return underlyingAmount The amount of underlying token with 8 decimal
+     */
+    function computeUnderlyingAmount(
+        address _token,
+        address _underlyingToken,
+        uint256 _amount
+    ) public returns (uint88) {
+        uint256 underlyingDecimals = ERC20(_underlyingToken).decimals();
+        if (_token != _underlyingToken) {
+            uint256 exchangeRateCurrent = CToken(_token).exchangeRateCurrent();
+            uint256 mantissa = 10 + underlyingDecimals;
+            uint256 underlyingAmount = (_amount * exchangeRateCurrent) / (10**mantissa);
+            return uint88(underlyingAmount);
+        }
+        if (underlyingDecimals >= 8) {
+            return uint88(_amount / 10**(underlyingDecimals - 8));
+        }
+        return uint88(_amount * 10**(8 - underlyingDecimals));
+    }
+
+    /**
+     * @notice Approve the fcash contract to use the input token and mint fcash token
+     * @dev If the input token is ETH, we need to deposit it into WETH. If the token is a ctoken,
+     * we need to call mintViaAsset. Otherwise, we just need to call mintViaUnderlying.
+     * @dev Will revert if the wrapped fcash contract has not been deployed
+     * @param _inputToken The address of the input asset
+     * @param _outputToken The address of the output asset
+     * @param _currencyId The predefined Notional currency id defined in constructor
+     * @param _amount The amount of input token
+     * @param _maturity The maturity of the market we want to enter
+     * @return outputAmount The amount of fcash token we actually receive
+     */
+    function _enter(
+        address _inputToken,
+        address _outputToken,
+        uint16 _currencyId,
+        uint256 _amount,
+        uint40 _maturity
+    ) internal returns (uint256) {
+        IWrappedfCash fcash = IWrappedfCash(FCASH_FACTORY.computeAddress(_currencyId, _maturity));
+        if (address(fcash) != _outputToken) {
+            revert ErrorLib.InvalidOutputA();
+        }
+        if (_inputToken == address(0)) {
+            WETH9(WETH).deposit{value: _amount}();
+            _inputToken = WETH;
+        }
+        if (!preapproved[_inputToken][address(fcash)]) {
+            ERC20(_inputToken).approve(address(fcash), type(uint256).max);
+            preapproved[_inputToken][address(fcash)] = true;
+        }
+        if (fcash.getCurrencyId() != _currencyId) {
+            revert("fcash has not been deployed");
```

Comment:
In what situations would this condition happen? If fcash was not deployed then you could not receive currencyId from it, no?